sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala

```diff
@@ -142,7 +142,7 @@ private[sql] object OrcFilters extends OrcFiltersBase {
     case BooleanType => PredicateLeaf.Type.BOOLEAN
     case ByteType | ShortType | IntegerType | LongType => PredicateLeaf.Type.LONG
     case FloatType | DoubleType => PredicateLeaf.Type.FLOAT
-    case StringType | _: CharType | _: VarcharType => PredicateLeaf.Type.STRING
+    case StringType => PredicateLeaf.Type.STRING
```

Comment:
Thank you so much for your swift fix, @cloud-fan . Do you think we can have a test coverage to prevent us from making this mistake again please?

Reply:
The test added by @yaooqinn should be good enough. AFAIK ORC is the only popular file format that supports CHAR/VARCHAR natively, and the only file format in Spark that we use a parser to convert its schema to catalyst schema.

For long-term maintainability, I think we should follow `ParquetToSparkSchemaConverter` and write a true schema converter for ORC, instead of using the parser.